lib/js/src/util/TextEncoder.js

```diff
@@ -0,0 +1,53 @@
+class TextEncoder {
+    // https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder#Polyfill
+    encode (str) {
```

Comment:
Looks like it doesn't make sense to require an instance of the class to access the `encode()` method. Recommend changing to:
```suggestion
    static encode (str) {
```

Reply:
updated to static